lib/fpm/package.rb

```diff
@@ -249,8 +257,20 @@ def output(path)
                                   "creating #{self.type} packages")
   end # def output
 
+  def temporary_directory(path)
+    if attributes[:source_date_epoch].nil?
+      return Stud::Temporary.directory("package-#{type}-staging")
+    else
+      # Choose a predictable build path; see https://reproducible-builds.org/docs/build-path/
+      root = ENV["TMP"] || ENV["TMPDIR"] || ENV["TEMP"] || "/tmp"
```

Comment:
Yes, the example from the next hunk down shows it.  Here it is with more context:
```
$ fpm -s gem -t deb json; mkdir run1; mv *.deb run1
$ sleep 1
$ fpm -s gem -t deb json; mkdir run1; mv *.deb run2
$ diffoscope run1/rubygem-run2/json_2.1.0_amd64.debrubygem-json_2.1.0_amd64.deb 
...
├── data.tar.gz
│   ├── data.tar
│   │   ├── ./var/lib/gems/2.3.0/extensions/x86_64-linux/2.3.0/json-2.1.0/json/ext/generator.so
│   │   │   ├── readelf --wide --notes {}
│   │   │   │ @@ -1,5 +1,5 @@
│   │   │   │  
│   │   │   │  Displaying notes found at file offset 0x000001c8 with length 0x00000024:
│   │   │   │    Owner                 Data size        Description
│   │   │   │    GNU                  0x00000014        NT_GNU_BUILD_ID (unique build ID bitstring)
│   │   │   │ -    Build ID: 438800c00798505fc92d9302ebf4ecfcd5aa5f57
│   │   │   │ +    Build ID: 6364fd6a8cad33df79bd85b47faa3617e13304fe
│   │   │   ├── readelf --wide --debug-dump=info {}
│   │   │   │ @@ -2,60 +2,60 @@
│   │   │   │  
│   │   │   │    Compilation Unit @ offset 0x0:
│   │   │   │     Length:        0x7605 (32-bit)
│   │   │   │     Version:       4
│   │   │   │     Abbrev Offset: 0x0
│   │   │   │     Pointer Size:  8
│   │   │   │   <0><b>: Abbrev Number: 1 (DW_TAG_compile_unit)
│   │   │   │ -    <c>   DW_AT_producer    : (indirect string, offset: 0x361): GNU C11 5.4.0 20160609 -mtune=generic -march=x86-64 -g -O2 -fstack-protector-strong -fPIC -fstack-protector-strong
│   │   │   │ +    <c>   DW_AT_producer    : (indirect string, offset: 0x2d1): GNU C11 5.4.0 20160609 -mtune=generic -march=x86-64 -g -O2 -fstack-protector-strong -fPIC -fstack-protector-strong
│   │   │   │      <10>   DW_AT_language    : 12        (ANSI C99)
│   │   │   │      <11>   DW_AT_name        : (indirect string, offset: 0x1147): generator.c
│   │   │   │ -    <15>   DW_AT_comp_dir    : (indirect string, offset: 0xae): /tmp/package-gem-staging-7f3a2a248718996b858ad69aa865ca5dc46125339c4d6f163bedf10f0b24/var/lib/gems/2.3.0/gems/json-2.1.0/ext/json/ext/generator
│   │   │   │ +    <15>   DW_AT_comp_dir    : (indirect string, offset: 0xe4a): /tmp/package-gem-staging-6d7699fb2e5ccddffbb0ca5b66855e3971f9db4da3984f8ef7845da10088/var/lib/gems/2.3.0/gems/json-2.1.0/ext/json/ext/generator
```